src/main/java/captain/cybot/adventure/backend/controller/UserController.java

```diff
@@ -0,0 +1,51 @@
+package captain.cybot.adventure.backend.controller;
+
+import captain.cybot.adventure.backend.exception.InvalidRoleException;
+import captain.cybot.adventure.backend.exception.PasswordInvalidException;
+import captain.cybot.adventure.backend.exception.UserAlreadyExistsException;
+import captain.cybot.adventure.backend.model.user.User;
+import captain.cybot.adventure.backend.service.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+
+@RestController
+@RequestMapping("/api/v0/users")
+@RequiredArgsConstructor
+@Slf4j
+public class UserController {
+
+    private final UserService userService;
+
+    @PostMapping("")
+    public ResponseEntity<?> create(@Valid @RequestBody User user) {
```

Comment:
Should this method handle setting of default USER role?

Reply:
It does but, in the UserService layer. Line 63 in the UserServiceImpl class in method saveUser(), this line adds the default user role to the user:
`addRoleToUser(newUser.getUsername(), ROLES.ROLE_USER.toString());`